test/Microsoft.AspNetCore.JsonPatch.Test/JsonPropertyComplexNameDTO.cs

```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Newtonsoft.Json;
+
+namespace Microsoft.AspNetCore.JsonPatch
+{
+    public class JsonPropertyComplexNameDTO
+    {
+        [JsonProperty("foo/bar~")]
```

Comment:
I've added a two theories to test directly the ParsedPath class. I also added another property on this test.

I've also added a test involving dictionaries (which was my initial issue actually).